src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs

```diff
@@ -655,13 +656,88 @@ protected override void PrepareContainerForItemOverride(DependencyObject element
 			}
 		}
 
+		private const string DragItemsFormatId = DataPackage.UnoPrivateDataPrefix + "__dragged__items__";
+
 		internal override void ContainerPreparedForItem(object item, SelectorItem itemContainer, int itemIndex)
 		{
 			base.ContainerPreparedForItem(item, itemContainer, itemIndex);
 
+			if (CanDragItems)
+			{
+				itemContainer.CanDrag = true;
+				itemContainer.DragStarting += OnItemContainerDragStarting;
+				itemContainer.DropCompleted += OnItemContainerDragCompleted;
+			}
+
 			ContainerContentChanging?.Invoke(this, new ContainerContentChangingEventArgs(item, itemContainer, itemIndex));
 		}
 
+		internal override void ContainerClearedForItem(object item, SelectorItem itemContainer)
```

Comment:
I don't think this will currently work the way you want it to. For ListView (with ItemsStackPanel or ItemsWrapGrid), `ItemsControl.CleanUpContainer()` is only called on Android, and even there it's not called every time an element is recycled. You'll end up with multiple drag subscriptions.

If you opt to fix this, I'd suggest to add a test in Uno.UI.RuntimeTests confirming that ClearContainerForItemOverride() is correctly raised in the same manner as UWP on all platforms.

Reply:
😱

Reply:
So I was a bit lazy ... I only made sure to unsubscribe events before re-subscribing. As handler are static it's enough.